### ISSUE TITLE
Make love types jumpable by wrapping with || instead of <>

### ISSUE THREAD

**Issue by idbrii** [davisdude/vim-love-docs#19]
Currently generates docs that look like:

```
            `align`: <AlignMode>

                The alignment.
```

AlignMode is defined elsewhere but pressing K on it says  "Sorry, no help for <AlignMode>". If you replace all <> with || (`%sm/\v: \<(.*)\>$/: |\1|`), then AlignMode becomes jumpable.

If lua types were prefixed with lrv- then they would be jumpable with [luarefvim](https://github.com/idbrii/luarefvim).

**Comment by idbrii** [comment]
[Here's this solution](https://github.com/idbrii/luarefvim/commit/4b1f703ab03112a2e8e01de3087362c2930de031) applied directly to the generated text. Instead, we should apply it to the generator.

**Comment by davisdude** [comment]
Hey, thanks for creating the issue! Could you check out the issue19 branch to see if that fixes it for you? Thanks

**Comment by idbrii** [comment]
Two issues:

# No links for love types

Looks like that branch resolves looks like it resolves lua built-in types, but not love types.

It still generates snippets like this:

```
        arguments:

            `joystick`: |Joystick|

                The joystick object.
```

Which don't jump to the definition because of the love- prefix:

```
-------------------------------------------------------------------------------
                                                                *love-Joystick*
|Joystick|
```



# Nontypes formatted as types

There are some things that aren't types that are formatted as types (there's no lrv-value or love-value).

```
    1.  `encoded` = |love.data.encode|( `container`, `format`,
        `sourceString`, `linelength` )

        See function description

        returns:

            `encoded`: |lrv-value|

                ByteData/string which contains the encoded version of source.
```

# Fix

Looks like this fixes both (LUA_TYPES is pulled from [the manual](http://www.lua.org/manual/5.4/manual.html#2.1) and those types are the same in 5.1 and 5.4):

```diff
M src/doc/main.lua
@@ -13,6 +13,17 @@ local TOC_NAME_REF_SPACING = 2
 local TOC_REF_WIDTH_LIMIT = PAGE_WIDTH - TOC_NAME_WIDTH_LIMIT - TOC_NAME_REF_SPACING
 
 local LOVE_TYPES = {}
+
+local LUA_TYPES = {
+    'boolean',
+    'function',
+    'nil',
+    'number',
+    'string',
+    'table',
+    'thread',
+    'userdata',
+}
 -- }}}
 
 -- Misc. functions {{{
@@ -58,9 +69,11 @@ end
 
 local function formatAsType( str )
 	if LOVE_TYPES[str] then
-		return ('|%s|'):format( str )
-	else
+		return ('|love-%s|'):format( str )
+	elseif LUA_TYPES[str] then
 		return ('|lrv-%s|'):format( str )
+    else
+		return ('<%s>'):format( str )
 	end
 end
```

**Comment by idbrii** [comment]
Three issues. I found solutions for 2/3, so I'll open a PR.

# No links for love types

Looks like that branch resolves looks like it resolves lua built-in types, but not love types.

It still generates snippets like this:

```
        arguments:

            `joystick`: |Joystick|

                The joystick object.
```

Which don't jump to the definition because of the love- prefix:

```
-------------------------------------------------------------------------------
                                                                *love-Joystick*
|Joystick|
```



# Nontypes formatted as types

There are some things that aren't types that are formatted as types (there's no lrv-value or love-value).

```
    1.  `encoded` = |love.data.encode|( `container`, `format`,
        `sourceString`, `linelength` )

        See function description

        returns:

            `encoded`: |lrv-value|

                ByteData/string which contains the encoded version of source.
```

# Aggregate or multiple possible types

Occasionally, there's more than one type specified.

```
                `...`: <tables and strings>

                `mipmaps`: <boolean or table>
```

**Comment by davisdude** [comment]
Fixed with commit 227e35bb50b54de25c12d05e6509774bc7bd8c83

In regards to problem 3, due to the way love-api formats parameters like this (see [here](https://github.com/love2d-community/love-api/blob/167efd11d31748e9dcb7f83d212178bd6978eab1/modules/graphics/Graphics.lua#L2259) for instance), I don't see much I can do here without implementing a hacky workaround that I'm not a fan of (like parsing for ` or ` in the type string).

After looking through the api, it appears that that is the only type like that.
